authorization/src/main/kotlin/org/modelix/authorization/AuthorizationConfig.kt

```diff
@@ -73,42 +77,64 @@ interface IModelixAuthorizationConfig {
      */
     var hmac256Key: String?
 
+    /**
+     * This key is made available at /.well-known/jwks.json so that other services can verify that a token was created
+     * by this server.
+     */
+    var ownPublicKey: JWK?
+
+    /**
+     * In addition to JWKS URLs you can directly provide keys for verification of tokens sent in requests to
+     * this server.
+     */
+    fun addForeignPublicKey(key: JWK)
+
     /**
      * If RSA signatures a used, the public key will be downloaded from this registry.
      */
     var jwkUri: URI?
 
     /**
-     * The ID of the public key for the RSA signature.
+     * If set, only this key is allowed to sign tokens, even if the jwkUri provides multiple keys.
      */
+    @Deprecated("Untrusted keys shouldn't even be return by the jwkUri or configured in some other way")
     var jwkKeyId: String?
 
     /**
      * Defines the available permissions and their relations.
      */
     var permissionSchema: Schema
 
+    /**
+     * Via /permissions/manage, users can grant permissions to ID tokens.
+     * By default, changes are not persisted.
+     * As an alternative to this configuration option, the environment variable MODELIX_ACCESS_CONTROL_FILE can be used
+     * to write changes to disk.
+     */
+    var accessControlPersistence: IAccessControlPersistence
+
     /**
      * Generates fake tokens and allows all requests.
      */
     fun configureForUnitTests()
 }
 
 class ModelixAuthorizationConfig : IModelixAuthorizationConfig {
-    override var permissionChecksEnabled: Boolean? = getBooleanFromEnv("MODELIX_PERMISSION_CHECKS_ENABLED")
+    override var permissionChecksEnabled: Boolean? = PERMISSION_CHECKS_ENABLED
     override var generateFakeTokens: Boolean? = getBooleanFromEnv("MODELIX_GENERATE_FAKE_JWT")
     override var debugEndpointsEnabled: Boolean = true
+    override var installStatusPages: Boolean = false
     override var hmac512Key: String? = null
     override var hmac384Key: String? = null
     override var hmac256Key: String? = null
-    override var jwkUri: URI? = System.getenv("MODELIX_JWK_URI")?.let { URI(it) }
-        ?: System.getenv("KEYCLOAK_BASE_URL")?.let { keycloakBaseUrl ->
-            System.getenv("KEYCLOAK_REALM")?.let { keycloakRealm ->
-                URI("${keycloakBaseUrl}realms/$keycloakRealm/protocol/openid-connect/certs")
-            }
-        }
+    override var ownPublicKey: JWK? = null
+    private val foreignPublicKeys = ArrayList<JWK>()
+    override var jwkUri: URI? = null
     override var jwkKeyId: String? = System.getenv("MODELIX_JWK_KEY_ID")
     override var permissionSchema: Schema = buildPermissionSchema { }
+    override var accessControlPersistence: IAccessControlPersistence = System.getenv("MODELIX_ACCESS_CONTROL_FILE")
```

Comment:
How do I support this, and also configuration via environment variables in Kubernetes?

Reply:
The documentation says
> To get access to configuration file properties, pass ApplicationConfig to the configuration class constructor.

If I understand it correctly, if have to explicitly implement support for this file based configuration, which wasn't done before and still isn't. I don't understand the problem with reading this environment variable.

Reply:
Ah, it would exist automatically if we were using engine main instead of launching the server manually. Then let's keep that for a follow up that refactors to launching via engine main.